### DISCUSSION TITLE
Deepin Movie: Pause playback when preview desktop / user move to other workspaces

### DISCUSSION THREAD

**Discussion by aiamuzz** [linuxdeepin/developer-center#3033] [Features Request & Ideas | 特性请求 & 头脑风暴]
Hi,

I have freshly installed DeepinOS 15.10.2 and have been testing stuff one by one ...

Deepin Movie until sometime back had a very nice feature which was that it would pause/stop when its window was minimized or if one moved to another workspace ... 

I don't know if this is bug or a part of WIP given the whole transitioning from unstable to stable ... and from deepin-wm to kwin !!!

I am just reporting it what ever the case may be so that it can be restored to its previous functionality which was a fantastic one !!!

Hope to see Deepin-Movie return to pausing when minimized or when a workspace is switched ...

Cheers

**Comment by BLumia** [comment]
Pause when minimized works on my machine with dde-kwin, It will not paused when swich workspace(if it does before, then it's more likely a bug).

**Comment by aiamuzz** [comment]
> Pause when minimized works on my machine with dde-kwin,

hmmmmm ... doesn't work for me !

> It will not paused when swich workspace(if it does before, then it's more likely a bug)

really ... a Bug ?
I thought pausing '**when one switched to another workspace**' was pretty consistent with the principle of '**pausing it when minimized**' ... **the principle being ...** that the user isn't watching the video when minimized or when he switches the workspace, hence pausing video makes sense !

all the DeepinOS versions from Debian Unstable were consistent with the behaviour of pausing when minimized and switching workspace !!!

@BLumia ... **You may want to double check on** ... "pausing when workspace is switched" being a bug or by design !!!

**Comment by BLumia** [comment]
Switch workspace won't cause minimize so it of course will not pause, unless the option means "pause when lost focus" or "pause when switch workspace".

I did a quick code search and seems there are no specific code to implement "pause when switch workspace", or maybe I missed somewhere?

**Comment by aiamuzz** [comment]
> I did a quick code search and seems there are no specific code to implement "pause when switch workspace", or maybe I missed somewhere?

@BLumia ... oh really ... i am still betting you may have missed it ... :smile: ...

A 'bug' which in principle aligns perfectly with the **feature of pausing when minimized**(as user will no longer be watching the video) must be the **"Best Bug Ever"** ... 

**My worst fear :**  by any chance, is it possible that Deepin code may have self spawned an AI ... which may have programmed this feature !!!
:laughing: !!!

Then its only natural that i will have to requisition the above bug to be reinstated as an officially supported feature in the new kwin(if its WM dependent) implementation or in the new Debian Stable based DeepinOS ...

**Comment by aiamuzz** [comment]
@BLumia ...

Hey ... sorry i think i was kind of liberal when i said its not pausing when its minimized ... turns out ... minimizing the movie window is totally a separate action then hitting `Meta+D` which minimize all application windows and shows the desktop 

Let me re-iterate ... previously Deepin Movie would pause when the following were performed ...

1. When one minimized the movie window (accounted for) - seems this is actually working for me too !
2. When we hit `Meta(Super)+D` which minimizes all windows and shows the desktop ... this is what i have reported as not working ... turns out i referred this action of hitting `Meta+D` as minimizing ...
3. When we switch the workspace the video used to pause.

So to summarize ... 2 behaviors out of 3 are not working as before ... and all 3 actions work on the same principle that in all the above 3 cases the user isn't watching the video and the video was paused ... So i guess it would be too much of a coincidence for 2 bugs to operate in principle to 1 intended action !!!

I really hope Deepin isn't infested by a self spawned Artificial Intelligence(AI) ...

Whoa ... if so ... then the world may be coming to an end and the machines are going to take over ... :smile: ... :smile: ... :smile:

**Comment by BLumia** [comment]
> Hey ... sorry i think i was kind of liberal when i said its not pausing when its minimized ...

It's okay :)

> turns out, minimizing the movie window is totally a separate action then hitting `Meta+D` which minimize all application windows and shows the desktop

`Meta+D` is "preview desktop", for applications it also won't get a notify about minimized since it doesn't get minimized, the actual behavior is when press `Meta+D`, it raise the x11 root window (the desktop) on the top.

**Comment by BLumia** [comment]
Then mark it as a feature request now.

cc @wangjia12

**Comment by aiamuzz** [comment]
> Then mark it as a feature request now.
> 
> cc @wangjia12

thanks good to have this functionality back ... but is it possible that it wasn't intended yet we had two unexpected bugs( ... an SOS sent to @**avengers_the_guardians_of_the_earth** !!! ... LinuxDeepin under AI attack)** that were in line with "pause video at instances where user isn't watching the video" ... which totally seemed they were as a matter of design !!!
